### ISSUE TITLE
Any plans to implement regexes in DSL-reconstruction?

### ISSUE THREAD

**Issue by tsbertalan** [lark-parser/lark#628]
https://github.com/lark-parser/lark/blob/54027942515054682a2958d7a7570a162311c177/lark/reconstruct.py#L38

I have an application in which code generation is as important as code parsing. I can generate correct DSL code as a repr of my own objects fairly easily (it's of course the parsing task for which Lark is really essential), but it would be nice to have a single source of truth for the DSL definition in my .lark file.

However, my grammar contains a few regexes (`_LABELMARK: /(?<!\\)@/`, `_HASHMARK: /(?<!\\)#/`, and the tutorials' `_NEWLINE: /(\r?\n[\t ]*)+/`), and so can't be used with the current `Reconstructor` interface. **Are there any plans to do some sort of "defualt guess" for these regexes?** I guess it's a pretty hard problem; I don't myself don't know of a general way to do this, hence the scare quotes.

**Comment by erezsh** [comment]
Making an automatic guess is possible, to the extent that the regex is perfectly restrictive.

I'll have to think about how difficult that would be.

But meanwhile, it's possible to expose an interface that will let you generate the text, based on the terminal name. Would that solve your problem?

**Comment by tsbertalan** [comment]
The mapping from strings to regex matchings is certainly not a bijection--you might have multiple documents that all elicit the same response from a regex parser. I believe that choosing a _default_ response should in principle be possible (select any one of those documents, perhaps the "simplest" one), but is probably difficult to do efficiently without built-in support from the underlying regex engine.

That said, it's far easier for me as the language designer to know ahead of time what that one "simplest" interpretation is; I already solved the problem for my own purposes by just writing an appropriate collection of `__str__` methods on the objects I generate from your `Transformer` interface. (Actually, I went further and decided that, for my application, it's better to just use Markdown+extensions than to define my own DSL, but that's not the point.)

I guess a nice middle ground would be to make something like the `Transformer` interface for going backwards from the `Tree` hierarchy (the result of `Lark.parse`) to a string representation. But only if there's enough demand, and a slightly deeper consideration of the issue than mine reveals a pattern more elegant than my quickly-hacked-together `__str__` technique. It's probably better for a library to be minimal to do the hard things like parsing, and let me handle the easy things myself.

So, I'd say, add it to the "someday" list and move on.
